scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,1038 @@
+                                          ", success=" + answer,
+                                 Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public boolean replace(String originator, TaskId taskid, String channel, String key, Serializable oldValue,
+            Serializable newValue) throws InvalidChannelException, IOException {
+        boolean answer = getChannel(channel).replace(key, oldValue, newValue);
+        logWithContextAndPersist(taskid,
+                                 channel,
+                                 "Tried to replace key " + QUOTE + key + QUOTE + " with expected old value " +
+                                          oldValue + " and new value " + newValue + ", success=" + answer,
+                                 Level.DEBUG);
+        return answer;
+
+    }
+
+    @Override
+    public Serializable replace(String originator, TaskId taskid, String channel, String key, Serializable value)
+            throws InvalidChannelException, IOException {
+        Serializable answer = getChannel(channel).replace(key, value);
+        logWithContextAndPersist(taskid,
+                                 channel,
+                                 "Replaced key " + QUOTE + key + QUOTE + WITH_VALUE + value + ", old value was " +
+                                          answer,
+                                 Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Serializable computeIfAbsent(String originator, TaskId taskid, String channel, String key,
+            String mappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        Serializable answer;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            answer = chosenChannel.computeIfAbsent(key,
+                                                   (Function<? super String, ? extends Serializable>) evaluateClosure(mappingFunction,
+                                                                                                                      Function.class));
+            logWithContextAndPersist(taskid,
+                                     channel,
+                                     "Run computeIfAbsent" + ON_KEY + QUOTE + key + QUOTE + RESULT_IS + answer,
+                                     Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Serializable computeIfPresent(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)